lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -1640,3 +1647,182 @@ def test_banner_display_off(self):
+
+
+class TestInCourseReverifyView(ModuleStoreTestCase):
+    """
+    Tests for the incourse reverification views.
+    """
+    IMAGE_DATA = "abcd,1234"
+    def setUp(self):
+        super(TestInCourseReverifyView, self).setUp()
+
+        self.user = UserFactory.create(username="rusty", password="test")
+        self.client.login(username="rusty", password="test")
+
+        self.course_key = SlashSeparatedCourseKey("Robot", "999", "Test_Course")
+        CourseFactory.create(org='Robot', number='999', display_name='Test Course')
+
+        # Create the course modes
+        for mode in ('audit', 'honor', 'verified'):
+            min_price = 0 if mode in ["honor", "audit"] else 1
+            CourseModeFactory(mode_slug=mode, course_id=self.course_key, min_price=min_price)
+
+        # Enroll the user in the default mode (honor) to emulate
+        CourseEnrollment.enroll(self.user, self.course_key, mode="verified")
+        self.config = InCourseReverificationConfiguration(enabled=True)
+        self.config.save()
+
+        # mocking and patching for bi events
+        analytics_patcher = patch('verify_student.views.analytics')
+        self.mock_tracker = analytics_patcher.start()
+        self.addCleanup(analytics_patcher.stop)
+
+    @patch.dict(settings.FEATURES, {'AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING': True})
+    def test_incourse_reverify_feature_flag_get(self):
+        self.config.enabled = False
+        self.config.save()
+        url = reverse('verify_student_incourse_reverify',
+                      kwargs={"course_id": unicode(self.course_key), "checkpoint_name": "midterm"})
+        response = self.client.get(url)
+
+        self.assertEquals(response.status_code, 404)
+
+    @patch.dict(settings.FEATURES, {'AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING': True})
+    def test_incourse_reverify_invalid_course_get(self):
+        url = reverse('verify_student_incourse_reverify',
+                      kwargs={"course_id": "invalid/course/key", "checkpoint_name": "midterm"})
+        response = self.client.get(url)
+
+        self.assertEquals(response.status_code, 404)
+
+    @patch.dict(settings.FEATURES, {'AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING': True})
+    def test_incourse_reverify_invalid_checkpoint_get(self):
+        url = reverse('verify_student_incourse_reverify',
+                      kwargs={"course_id": unicode(self.course_key), "checkpoint_name": "invalid_checkpoint"})
+        response = self.client.get(url)
+
+        self.assertEquals(response.status_code, 404)
+
+    @patch.dict(settings.FEATURES, {'AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING': True})
+    def test_incourse_reverify_initial_redirect_get(self):
+        self._create_checkpoint()
+        url = reverse('verify_student_incourse_reverify',
+                      kwargs={"course_id": unicode(self.course_key), "checkpoint_name": "midterm"})
+        response = self.client.get(url)
```

Comment:
Since you use this URL construction and request-issuing pattern frequently throughout your tests, I'd recommend abstracting it with a private helper method.

Also, constant values used frequently in these tests, like mode names and "midterm", should be assigned to variables.